connection.go

```diff
@@ -512,7 +534,10 @@ func (c *Connection) protocolError(id uint32, err error) error {
 	sysErr := NewWrappedSystemError(ErrCodeProtocol, err)
 	c.SendSystemError(id, Span{}, sysErr)
 	// Don't close the connection until the error has been sent.
-	c.Close()
+	c.close(LogFields{
```

Comment:
is there a reason to create the `LogFields{}` container at the call site rather than having the compiler do whatever it feels like with:

```go
c.close(
  LogField{"reason", "protocol error"},
  ErrField(err),
)
```

syntactic sugar in case another `LogField{}` is created in-place?